internal/controller/install/common_helpers.go

```diff
@@ -317,7 +317,16 @@ func ExtractPulsarConfig(config runtime.RawExtension) (PulsarConfig, error) {
 	if err != nil {
 		return PulsarConfig{}, err
 	}
-	return asConfig.Pulsar, nil
+	return asConfig.Pulsar, err
```

Comment:
is there a need to return `err` as it will 100% be `nil`?

Reply:
No I can change that